src/transactions/api.ts

```diff
@@ -24,6 +29,57 @@ export const transactionFeedV2Api = createApi({
       keepUnusedDataFor: 60, // 1 min
     }),
   }),
+
+  /**
+   * Typed "any" as per the docs:
+   * https://redux-toolkit.js.org/rtk-query/usage/persistence-and-rehydration#redux-persist
+   *
+   * Implementation is based on this conversation:
+   * https://github.com/reduxjs/redux-toolkit/issues/4334
+   */
+  extractRehydrationInfo: (action): any => {
+    if (isRehydrateAction(action)) {
+      const persistedWeb3 = getRehydratePayload(action, 'web3') as RootState['web3'] | undefined
+      const walletAddress = persistedWeb3?.account?.toLowerCase() ?? null
+
+      const persistedTransactions = getRehydratePayload(action, 'transactions') as
+        | RootState['transactions']
+        | undefined
+      const feedFirstPage = persistedTransactions?.feedFirstPage || []
+
+      if (walletAddress && feedFirstPage.length) {
+        const queryKey = `transactionFeedV2({"address":"${walletAddress}","endCursor":${FIRST_PAGE_TIMESTAMP}})`
+        return {
+          mutations: {},
+          provided: {},
+          subscriptions: {},
+          queries: {
+            [queryKey]: {
+              status: QueryStatus.fulfilled,
+              endpointName: 'transactionFeedV2',
+              requestId: nanoid(),
+              originalArgs: { address: walletAddress, endCursor: FIRST_PAGE_TIMESTAMP },
+              startedTimeStamp: Date.now(),
+              data: { transactions: feedFirstPage },
+              fulfilledTimeStamp: Date.now(),
+              error: undefined,
+            },
+          },
+          config: {
+            online: true,
+            focused: true,
+            middlewareRegistered: true,
+            refetchOnFocus: false,
+            refetchOnReconnect: false,
+            refetchOnMountOrArgChange: false,
+            keepUnusedDataFor: 60,
+            reducerPath: 'transactionFeedV2Api',
+            invalidationBehavior: 'delayed',
+          },
+        } satisfies RootState['transactionFeedV2Api']
```

Comment:
I guess this works, but it feels like it's quite specific to underlying details of RTK-Query.
And could break easily with updates.
What do you think of doing something like `const currentData = data.transactions || feedFirstPage` at the component instead?

Reply:
@jeanregisser
When RTK-Query fetches the data - it first casts it to `undefined` and after the request is finished populates it with the data from the response. This means that on every page load there will always be an extra re-render where `data?.transactions || feedFirstPage` will evaluate to `feedFirstPage` as `data` would be still `undefined` for the time of fetching. And this will subsequently trigger the merge of every page with the `feedFirstPage`. 
I'm sure there is a solution that will work that might be similar to what you've proposed (maybe with the `init` ref flag and `useEffect`) but this might introduce more complexity as there's gonna be a new issue of distinguishing between initial first page update and subsequent polling updates.

If there's gonna be an update to the library via `renovate` - the `satisfies RootState['transactionFeedV2Api']` clause will ensure that the errors will be thrown if the underlying structure changes which would force us to update our code alongside the library update.

But the caveat is real indeed, as this creates an extra point of maintenance. Although, from the info I've gathered from the Github issues of RTK-Query related specifically to usage with persisted storage and offline usage - this seems to be the recommended approach by the RTK-Query team themselves. I put the link to one of the issues in the comment above the implementation as it might provide a bit more context.